core/templates/pages/participation-playbook/playbook.component.spec.ts

```diff
@@ -0,0 +1,157 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the teach page.
+ */
+
+import { NO_ERRORS_SCHEMA, Pipe, EventEmitter } from '@angular/core';
+import { ComponentFixture, TestBed, async, fakeAsync, tick } from
+  '@angular/core/testing';
+
+import { PlaybookPageComponent } from './playbook.component';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { I18nLanguageCodeService } from 'services/i18n-language-code.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { TranslateService } from 'services/translate.service';
+
+@Pipe({name: 'translate'})
+class MockTranslatePipe {
+  transform(value: string, params: Object | undefined):string {
+    return value;
+  }
+}
+
+class MockTranslateService {
+  languageCode = 'es';
+  use(newLanguageCode: string): string {
+    this.languageCode = newLanguageCode;
+    return this.languageCode;
+  }
+}
+
+class MockI18nLanguageCodeService {
+  codeChangeEventEmiiter = new EventEmitter<string>();
+  getCurrentI18nLanguageCode() {
+    return 'en';
+  }
+
+  get onI18nLanguageCodeChange() {
+    return this.codeChangeEventEmiiter;
+  }
+}
+
+// Mocking window object here because changing location.href causes the
+// full page to reload. Page reloads raise an error in karma.
+class MockWindowRef {
+  _window = {
+    location: {
+      _hash: '',
+      _hashChange: null,
+      _href: '',
+      get hash() {
+        return this._hash;
+      },
+      set hash(val) {
+        this._hash = val;
+        if (this._hashChange === null) {
+          return;
+        }
+        this._hashChange();
+      },
+      get href() {
+        return this._href;
+      },
+      set href(val) {
+        this._href = val;
+      },
+      reload: (val) => val
+    },
+    get onhashchange() {
+      return this.location._hashChange;
+    },
+
+    set onhashchange(val) {
+      this.location._hashChange = val;
+    }
+  };
+  get nativeWindow() {
+    return this._window;
+  }
+}
+
+class MockSiteAnalyticsService {
+  registerApplyToTeachWithOppiaEvent(): void {
+    return;
+  }
+}
+
+let component: PlaybookPageComponent;
+let fixture: ComponentFixture<PlaybookPageComponent>;
+
+describe('Teach Page', function() {
+  let windowRef: MockWindowRef;
+  let siteAnalyticsService = null;
+
+  beforeEach(async(() => {
+    windowRef = new MockWindowRef();
+    TestBed.configureTestingModule({
+      declarations: [PlaybookPageComponent, MockTranslatePipe],
+      providers: [
+        {
+          provide: I18nLanguageCodeService,
+          useClass: MockI18nLanguageCodeService
+        },
+        { provide: SiteAnalyticsService, useClass: MockSiteAnalyticsService },
```

Comment:
Maybe keep the same indentation as the above lines? Like break after { and before }

Reply:
done